client/js/app/components/explorer/query_builder/group_by_field.js

```diff
@@ -24,25 +24,79 @@ var GroupByField = React.createClass({
     return this.props.value;
   },
 
+  handleChange: function(name, value) {
+    var newVal = this.props.value.slice();
+    var index = name.split('.')[1];
+    newVal[index] = value;
+    this.props.handleChange(name.split('.')[0], newVal);
+  },
+
+  toggleMultiGroupBy: function(event) {
+    event.preventDefault();
+    var newVal;
+    switch (this.props.value.length) {
+      case 0:
+        newVal = this.props.value = ['', ''];
+        break;
+      case 1:
+        newVal = this.props.value.concat(['']);
+        break;
+      case 2:
+        newVal = this.props.value.slice(0,1);
+        break;
+    }
+    this.props.handleChange('group_by', newVal);
+  },
+
   // React methods
 
   render: function() {
+    var secondGroupByField;
+    var toggleText = "Group by a second property";
+    var toggleIcon = "plus";
+
+    if (this.props.value.length > 1) {
```

Comment:
how would you feel about pulling out the logic to generate the select fields and the multigroup toggle in a different private method?